### DISCUSSION TITLE
Override service name

### DISCUSSION THREAD

**Discussion by samragu** [rabbitmq/cluster-operator#1550] [Q&A]
Is it possible to override the service name? cluster operator does not seem to expose metadata: name as an overridden field? We had deployed a RabbitMQ cluster using a custom statefulset definition (without using the operator) and had used a service name that microservices are using to connect to RabbitMQ. We are now replacing RabbitMQ cluster using the operator but would like to avoid cluster name to be the service name. Would like to name the cluster something but still use current service name

**Comment by Zerpet** [comment]
You could name your `RabbitmqCluster` something that does not match with the current service, and update your service selector to match the new Pods.

**Reply by mkuratczyk** [reply]
I think an `ExternalName` Service could also do the trick - the actual service would now be managed by the operator, but you can have an additional service with the old name, that points to the new service.
https://kubernetes.io/docs/concepts/services-networking/service/#externalname
